### PR TITLE
Implement toggle between card grid view and horizontal view

### PR DESCRIPTION
**Pull request by wei2912**
This PR tracks changes to implement a toggle as per https://github.com/AdvisorySG/mentorship-page/issues/468.

### CODE REVIEW

**Comment by wei2912** [comment]
Do note that upon merging, the website needs to be monitored for up to an hour to ensure the lambda deployment is working fine and the deployment can run successfully.

**Comment by wei2912** [comment]
Possible icons that can be used:
1. Card/Grid view: https://mui.com/material-ui/material-icons/?query=grid&selected=GridView
2. List view: https://mui.com/material-ui/material-icons/?query=list&selected=ViewList

Modal should be just the detailed card but centered with bio expanded, mb enlarged a bit, and with a close button.

Number of results per page should be fine tuned at the end.

File: src/components/ResultView.tsx [hunk 1]
```diff
@@ -97,11 +111,12 @@ const ResultView = ({
     organisation,
     role,
     school,
+    thumbnail_image_url,
   } = result;
 
   const displayCourseOfStudy =
     courseOfStudy && courseOfStudy.raw
-      ? fillHighlights(courseOfStudy.snippet, courseOfStudy.raw)
+      ? fillEllipsis(courseOfStudy.snippet, courseOfStudy.raw)
```
Comment (wei2912) [inline src/components/ResultView.tsx:119]: This should be reverted to `fillHighlights`.

**Comment by wei2912** [comment]
On smaller screens, the following piece of code causes scroll behaviour to break:

image

image

However, this fix causes part of the image to be cropped off in the modal on smaller screens, due to `align-items: center`:

image

This can be changed to `align-items: flex-start` but would cause the modal to be uncentered on larger screens:

image

File: src/components/ResultViewGrid.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        onClose={() => setIsModalOpen(false)}
+        aria-labelledby="modal-title"
+        aria-describedby="modal-description"
+        style={{
+          display: "flex",
+          alignItems: "flex-start",
+          margin: "auto",
+          maxWidth: "800px",
+          maxHeight: "80%",
+          overflow: "scroll",
```
Comment (Hackin7) [inline src/components/ResultViewGrid.tsx:94]: Should change this from scroll to auto, such that unnecessary scroll bars don't show up everywhere

Sample Test

image

image

**Review by Hackin7 — COMMENTED** [review]
Overall I think it is ok to merge this in, though there are some things which should be tweaked

image

image

# Autoflow issues

Some formatting does not look right. This probably should and can be fixed soon to improve the UI. This could be done by setting overflow to auto in ResultViewGrid.tsx line 94

image

Another thing is the Read Less... text. I think it should be hidden, but that's not a priority


# Minor Issues -> Can merge this one first and open new issues

Maybe in this format we can just auto set it to card? But this issue isn't that severe, and I think its ok if we left it that way and open another issue for this

image

image


# Code

Though I see there are quite some changes to the code,  the code itself and the CSS looks fine, and shouldn't have any conflicts

**Pull request merged**